qiskit/circuit/quantumcircuit.py

```diff
@@ -441,21 +441,26 @@ def calibrations(self, calibrations: dict):
         """
         self._calibrations = defaultdict(dict, calibrations)
 
-    def has_calibration_for(self, instruction: CircuitInstruction):
+    def has_calibration_for(self, instruction: CircuitInstruction | tuple):
         """Return True if the circuit has a calibration defined for the instruction context. In this
         case, the operation does not need to be translated to the device basis.
         """
-        if not self.calibrations or instruction.operation.name not in self.calibrations:
+        if isinstance(instruction, CircuitInstruction):
+            operation = instruction.operation
+            qubits = instruction.qubits
+        else:
+            operation, qubits, _ = instruction
+        if not self.calibrations or operation.name not in self.calibrations:
             return False
-        qubits = tuple(self.qubits.index(qubit) for qubit in instruction.qubits)
+        qubits = tuple(self.qubits.index(qubit) for qubit in qubits)
```

Comment:
Should this use find bits? It's the same as before so I'm fine merging it as is, but I just worry the performance of this is pretty poor. Although I guess the number of qubits for a gate won't be that wide so it's probably not a big deal.

Reply:
Yeah it probably should - this is doing unnecessary linear searches now.  I think there's a few places throughout Qiskit where we're doing `list.index`, and they're pretty much all potential performance problems.

Reply:
Well lets do all of them in a single pass, the performance here isn't critical for this PR, it's more to fix the api breakage.